cpp/tests/strings/floats_tests.cpp

```diff
@@ -125,11 +125,15 @@ TEST_F(StringsConvertTest, FromFloats32)
 
 TEST_F(StringsConvertTest, ToFloats64)
 {
+  // clang-format off
   std::vector<const char*> h_strings{
     "1234",   nullptr,    "-876",     "543.2",         "-0.12",   ".25",
     "-.002",  "",         "-0.0",     "1.28e256",      "NaN",     "abc123",
     "123abc", "456e",     "-1.78e+5", "-122.33644782", "12e+309", "1.7976931348623159E308",
-    "-Inf",   "-INFINITY"};
+    "-Inf",   "-INFINITY", "1.0",     "1.7976931348623157e+308",  "1.7976931348623157e-307",
+    // subnormal numbers
+    "4e-308", "3.3333333333e-320", "4.940656458412465441765688e-324", "1.e-324" };
```

Comment:
I would add a comment to indicate that the second to last one is the smallest double, and the last one is expected to parse as 0.